### PR TITLE
feat: remove substrafl wheel cache

### PR DESCRIPTION
**Pull request by SdgJlbl**
## Related issue

Fixes FL-1166

## Please check if the PR fulfills these requirements

- [ ] If the feature has an impact on the user experience, the [changelog](https://github.com/substra/substrafl/blob/main/CHANGELOG.md) has been updated
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] The commit message follows the [conventional commit](https://www.conventionalcommits.org/en/v1.0.0/) specification

### CODE REVIEW

**Comment by Owlfred** [comment]
End to end tests: :hourglass_flowing_sand: **STARTING**

**Comment by Owlfred** [comment]
End to end tests: :hourglass_flowing_sand: **STARTING**

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,6 +5,12 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## \[Unreleased\]
```
Comment (ThibaultFy) [inline CHANGELOG.md:8]: 2 times `Unreleased`

File: substrafl/dependency/manage_dependencies.py [hunk 2]
```diff
@@ -96,26 +96,19 @@ def local_lib_wheels(lib_modules: List[ModuleType], *, dest_dir: Path) -> List[s
...
+        # installed in 'dist' and take them from there.
+        # sys.executable takes the current Python interpreter instead of the default one on the computer
+        extra_args: list = []
+        if lib_name == "substrafl":
+            extra_args = [
+                "--find-links",
+                dest_dir.parent / "substra",
+                "--find-links",
+                dest_dir.parent / "substratools",
+            ]
```
Comment (ThibaultFy) [inline substrafl/dependency/manage_dependencies.py:109]: feel free to ignore
```suggestion
          extra_args = [
              "--find-links",
              dest_dir.parent / "substra",
              "--find-links",
              dest_dir.parent / "substratools",
          ]   if lib_name == "substrafl" else []
```

**Comment by Owlfred** [comment]
End to end tests: :hourglass_flowing_sand: **STARTING**

**Review by ThibaultFy — COMMENTED** [review]
Thx for the PR ! just waiting for the e2e tests to run :)

File: CHANGELOG.md (hunk 1 above)
Comment (SdgJlbl) [reply CHANGELOG.md:8]: In anticipation of the rebase following the upcoming release

**Review by ThibaultFy — APPROVED** [review]
CI ✅

**Comment by Owlfred** [comment]
End to end tests: :hourglass_flowing_sand: **STARTING**

**Comment by Owlfred** [comment]
Invalid repository in --refs: "'substra-backend". Supported repositories: hlf-k8s, orchestrator, substra-backend, substra-frontend, substra-tools, substrafl, substra, substra-tests, substra-documentation, substra-ci

**Comment by Owlfred** [comment]
End to end tests: :hourglass_flowing_sand: **STARTING**

**Comment by Owlfred** [comment]
```
Usage: /e2e [options] [help]

/e2e may appear anywhere as long as it is on its own line

Options:
  --refs <value>                                         Extra refs (branch or tag) with format REPO=GIT_REF,REPO=GIT_REF.
  Supported repositories: hlf-k8s, orchestrator, substra-backend, substra-frontend, substra-tools, substrafl, substra, substra-tests, substra-documentation, substra-ci.
  Example: /e2e --refs substra-backend=some_branch,orchestrator=some_tag (default: {})
  --tests-to-run, --tests <tests-to-run>                 Comma-separated list of tests to run. Valid options: sdk,substrafl,doc,frontend or NONE. (default: "sdk")
  --benchmarks-to-run, --benchmarks <benchmarks-to-run>  Comma-separated list of workflows tests to run. Valid options: mnist, camelyon or NONE. (default: "NONE")
  --orchestrator-mode, --mode <orchestrator-mode>        Comma-separated list of orchestrator modes to run tests for. Valid options: standalone,distributed (default: "standalone,distributed")
  -h, --help                                             display help for command
```

**Comment by Owlfred** [comment]
End to end tests: :hourglass_flowing_sand: **STARTING**

**Pull request merged**